src/node_native_module.cc

```diff
@@ -226,101 +229,111 @@ Local<Value> NativeModule::Compile(Environment* env,
         ScriptCompiler::CreateCodeCacheForFunction(fun));
     CHECK_NE(cached_data, nullptr);
-    char* data =
-        reinterpret_cast<char*>(const_cast<uint8_t*>(cached_data->data));
-
-    // Since we have no API to create a buffer from a new'ed pointer,
-    // we will need to copy it - but this code path is only run by the
-    // tooling that generates the code cache to be bundled in the binary
+    size_t cached_data_length = cached_data->length;
+    // Since we have no special allocator to create an ArrayBuffer
+    // from a new'ed pointer, we will need to copy it - but this
+    // code path is only run by the tooling that generates the code
+    // cache to be bundled in the binary
     // so it should be fine.
-    Local<Object> buf =
-        Buffer::Copy(env, data, cached_data->length).ToLocalChecked();
-    return scope.Escape(buf);
+    MallocedBuffer<uint8_t> copied(cached_data->length);
+    memcpy(copied.data, cached_data->data, cached_data_length);
+    Local<ArrayBuffer> buf =
+        ArrayBuffer::New(isolate,
+                         copied.release(),
+                         cached_data_length,
+                         ArrayBufferCreationMode::kInternalized);
+    return scope.Escape(Uint8Array::New(buf, 0, cached_data_length));
   } else {
     return scope.Escape(fun);
   }
 }
 
-void Initialize(Local<Object> target,
-                Local<Value> unused,
-                Local<Context> context) {
+void NativeModuleLoader::Initialize(Local<Object> target,
+                                    Local<Value> unused,
+                                    Local<Context> context) {
   Environment* env = Environment::GetCurrent(context);
+  Isolate* isolate = env->isolate();
 
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "source"),
-            env->native_modules_source())
-      .FromJust();
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "sourceHash"),
-            env->native_modules_source_hash())
-      .FromJust();
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "codeCache"),
-            env->native_modules_code_cache())
-      .FromJust();
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "codeCacheHash"),
-            env->native_modules_code_cache_hash())
-      .FromJust();
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "compiledWithCache"),
-            env->native_modules_with_cache())
-      .FromJust();
-  target
-      ->Set(context,
-            FIXED_ONE_BYTE_STRING(env->isolate(), "compiledWithoutCache"),
-            env->native_modules_without_cache())
-      .FromJust();
-
-  env->SetMethod(target, "compileFunction", NativeModule::CompileFunction);
-  env->SetMethod(target, "compileCodeCache", NativeModule::CompileCodeCache);
+  CHECK(target
+            ->SetAccessor(env->context(),
```

Comment:
I’d make this a method, as it signals that getting the value is a relatively expensive operation (and that the returned object is different every time this is called).

Ditto for `GetCacheUsage()`.